app/components/layouts/reducer.tsx

```diff
@@ -1,21 +1,29 @@
-import { LayoutState, LAYOUT_INITIAL_STATE, UserDevice } from './records';
-import { ACTION_TYPES } from '../../actions/actionTypes';
+import { createSlice, PayloadAction } from 'redux-starter-kit';
 
-export function reducer(state: LayoutState = LAYOUT_INITIAL_STATE, action: ReduxAction<any>): LayoutState {
-  switch (action.type) {
-    case ACTION_TYPES.SET_DEVICE_TO_DESKTOP: {
-      return { ...state, userDevice: UserDevice.DESKTOP };
-    }
+export enum UserDevice {
```

Comment:
FYI, I recently learned how to use enums within ambient types. Let me know if you think it would help you!

Reply:
sure. it would be helpful!

Reply:
Assuming your webpack is set up correctly, we just need to use the `const enum`. 

Since d.ts files are not included in the bundle, a regular enum in a d.ts, which gets compiled to an object, will disappear with the rest of the d.ts file. But a `const enum` gets compiled to constant values (like `1`) so the bundle will work regardless.

Reply:
I mention webpack because, in my case, the order of ts and js loaders somehow mattered. But you only use a TS loader, so it should be okay 🤞

Reply:
nice. thanks to ur comment, I've read https://www.typescriptlang.org/docs/handbook/enums.html#const-enums 
I'll apply it soon. thanks for your attention!